### DISCUSSION TITLE
Salesforce Integration broken after every deploy

### DISCUSSION THREAD

**Discussion by martinherweg** [verbb/formie#1140] [General] [answered]
Hey,

i'm not sure if this is a bug so I thought I'll open a discussion first.
We have a formie installation with a salesforce integration currently in sandbox mode, this works fine until a new deployment happens, after that the integration says it's still connected but when I refresh the integration within a form I get a curl error.

So I debugged it locally and found out the the `apiDomain` variable within the `Salesforce.php` is null when this error happens, when I reconnect the integration the `apiDomain` variable is set and everything works fine.

Would anyone have a hint why it could break after deploying? Is there some kind of token mismatch happening or something like this? Or is it even something that is on the Salesforce side?

Thanks and regards,
Martin

**Comment by engram-design** [comment, marked as answer]
Yep, so the `apiDomain` is [fetched and stored](https://github.com/verbb/formie/blob/ef8c29d875397a71be215a0e77f227b237fce83e/src/integrations/crm/Salesforce.php#L142-L150) when you first make the connection to Salesforce for the OAuth handshake. This is saved in the `settings` column in your `formie_integrations` database table. Are you able to have a look at this after a deploy, and it's no longer working?

Clearing any form of cache (which is normal for deployments) shouldn't have anything to do with this value disappearing.

The only thing I can think of is that your local Salesforce integration config (stored in project config) is somehow overwriting the production database settings on deploy. Downloading the production database and running `./craft project-config/rebuild` should ensure your local PC files are up to date as well.

**Reply by martinherweg** [reply]
That's exactly what happend, thanks for the fast reply!

So I guess the better option is to make the connection locally so the right domain is written to the project config?

**Reply by engram-design** [reply]
Short-term, yep! But this will likely need to be fixed properly by somehow merging the two settings (local and production) properly to ensure this doesn't happen down the line. I'll convert this to an issue to track.
